packages/blade/src/components/ProgressBar/ProgressBar.tsx

```diff
@@ -0,0 +1,185 @@
+   */
+  accessibilityLabel?: string;
+  /**
+   * Sets the contrast for the progress bar
+   * @default 'low'
+   */
+  contrast?: ColorContrastTypes;
+  /**
+   * Sets the intent of the progress bar which changes the feedback color.
+   */
+  intent?: Feedback;
+  /**
+   * Sets the label to be rendered for the progress bar. This value will also be used as default for `accessibilityLabel`.
+   */
+  label?: string;
+  /**
+   * Sets the size of the progress bar.
+   * @default 'small'
+   */
+  size?: 'small' | 'medium';
+  /**
+   * Sets the progress value of the progress bar.
+   * @default 'small'
+   */
+  value?: number;
+};
+
+type ProgressBarVariant = 'progress' | 'meter';
+
+type ProgressBarProgressProps = ProgressBarCommonProps & {
+  /**
+   * Sets the variant to be rendered for thr progress bar.
+   * @default 'progress'
+   */
+  variant?: Extract<ProgressBarVariant, 'progress'>;
+  /**
+   * Sets whether the progress bar is in an indeterminate state.
+   * @default false
+   */
+  isIndeterminate?: boolean;
+  /**
+   * Sets whether or not to show the progress percentage for the progress bar. Percentage is hidden by default for the `meter` variant.
+   * @default false
+   */
+  hidePercentage?: boolean;
+};
+
+type ProgressBarMeterProps = ProgressBarCommonProps & {
+  /**
+   * Sets the variant to be rendered for thr progress bar.
+   * @default 'progress'
+   */
+  variant?: Extract<ProgressBarVariant, 'meter'>;
+  /**
+   * Sets whether the progress bar is in an indeterminate state.
+   * @default false
+   */
+  isIndeterminate?: undefined;
+  /**
+   * Sets whether or not to show the progress percentage for the progress bar. Percentage is hidden by default for the `meter` variant.
+   * @default false
+   */
+  hidePercentage?: undefined;
+};
+
+type ProgressBarProps = ProgressBarProgressProps | ProgressBarMeterProps;
+
+type ProgressBarUnfilledProps = {
+  backgroundColor: string;
+};
+
+const ProgressBarUnfilled = styled(Box)<ProgressBarUnfilledProps>(({ backgroundColor }) => ({
+  backgroundColor,
+}));
+
+const getProgress = (value: number): number => {
+  return Math.floor(Math.min(100, Math.max(0, value)));
+};
+
+const progressBarHeight: Record<NonNullable<ProgressBarCommonProps['size']>, 2 | 4> = {
+  small: 2,
+  medium: 4,
+};
+
+const ProgressBar = ({
+  accessibilityLabel,
+  contrast = 'low',
+  intent,
+  isIndeterminate,
+  label,
+  hidePercentage = false,
+  size = 'small',
+  value = 0,
+  variant = 'progress',
+}: ProgressBarProps): ReactElement => {
+  console.log(isIndeterminate); // will be in follow up PR
+
+  if (variant === 'meter' && isIndeterminate) {
+    console.warn(`[Blade: ProgressBar]: Cannot set 'isIndeterminate' when 'variant' is 'meter'.`);
+  }
+
+  if (variant === 'meter' && hidePercentage) {
+    console.warn(
+      `[Blade: ProgressBar]: Cannot set 'hidePercentage' when 'variant' is 'meter'. Percentage is always hidden for meter.`,
+    );
+  }
+
+  const { theme } = useTheme();
```

Comment:
Suggestion: Ideally we should be invoking all hooks before any conditionals. If someone adds a conditional throw in either of the conditions above it could lead to weird rendering issues because hooks are meant to be invoked unconditionally. Not sure if our linting rules catch that right now.

Reply:
Done